### PR TITLE
Some Mech Things

### PR DESCRIPTION
**Pull request by Geevies**
Redoes how mechs assign their bodyparts and hardpoints. Should be much easier to make new mechs now, mechs can also now handle having children. Bless.
Created the backbone for having a mounted autolathe, however it needs a mech_equipment sprite, and a new GUI, which isn't something I'm capable of doing.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -15,10 +15,11 @@
...
 
-	var/hacked = 0
-	var/disabled = 0
-	var/shocked = 0
-	var/busy = 0
+	var/hacked = FALSE
+	var/disabled = FALSE
+	var/shocked = FALSE
+	var/busy = FALSE
+	var/build_item
```
Comment (mikomyazaki) [inline code/game/machinery/autolathe.dm]: Is there a point to having both a `build_item` and a `making` var? Why not just make `build_item` do all the stuff `making` does.

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -101,9 +110,9 @@
 					if(max_sheets && max_sheets > 0)
```
Comment (mikomyazaki) [inline code/game/machinery/autolathe.dm]: Seems redundant.

File: code/modules/heavy_vehicle/premade/powerloader.dm [hunk 3]
```diff
@@ -138,30 +121,20 @@
...
-	if(!body)
-		body = new /obj/item/mech_component/chassis/ripley(src)
-		body.color = "#849bc1"
-
-		body.mech_armor = new /obj/item/robot_parts/robot_component/armor/mech/combat(src)
+	e_head = /obj/item/mech_component/sensors/combatripley
+	e_body = /obj/item/mech_component/chassis/ripley
+	e_arms = /obj/item/mech_component/manipulators/ripley
+	e_legs = /obj/item/mech_component/propulsion/ripley
+	e_color = "#849bc1"
```
Comment (mikomyazaki) [inline code/modules/heavy_vehicle/premade/powerloader.dm]: You defined this colour but then didn't use the define.

File: code/game/machinery/autolathe.dm [hunk 4]
```diff
@@ -233,32 +233,31 @@
 	if(href_list["make"] && machine_recipes)
 		var/index = text2num(href_list["make"])
 		var/multiplier = text2num(href_list["multiplier"])
-		var/datum/autolathe/recipe/making
+		build_item = null
 
 		if(index > 0 && index <= machine_recipes.len)
```
Comment (mikomyazaki) [inline code/game/machinery/autolathe.dm]:
```suggestion
		if(index in 0 to machine_recipes.len)
```

Imo this is more readable. Can change in other places like line 242 too if you prefer this.

File: code/game/machinery/autolathe.dm [hunk 5]
```diff
@@ -233,32 +233,31 @@
...
-			making = machine_recipes[index]
+			build_item = machine_recipes[index]
 
 		//Exploit detection, not sure if necessary after rewrite.
-		if(!making || multiplier < 0 || multiplier > 100)
+		if(!build_item || multiplier < 0 || multiplier > 100)
```
Comment (mikomyazaki) [inline code/game/machinery/autolathe.dm]:
```suggestion
		if(!build_item || !(multiplier in 0 to 100))
```

File: code/game/machinery/autolathe.dm [hunk 6]
```diff
@@ -235,11 +235,11 @@
 		var/multiplier = text2num(href_list["multiplier"])
 		build_item = null
 
-		if(index > 0 && index <= machine_recipes.len)
+		if(index in 0 to machine_recipes.len)
```
Comment (mikomyazaki) [inline code/game/machinery/autolathe.dm]:
```suggestion
		if(index in 1 to machine_recipes.len)
```

Did you mean to include 0? It isn't equivalent to what it was before.

File: code/game/machinery/autolathe.dm [hunk 7]
```diff
@@ -184,90 +190,89 @@
...
 	if(href_list["make"] && machine_recipes)
-
 		var/index = text2num(href_list["make"])
 		var/multiplier = text2num(href_list["multiplier"])
-		var/datum/autolathe/recipe/making
+		build_item = null
 
-		if(index > 0 && index <= machine_recipes.len)
-			making = machine_recipes[index]
+		if(index in 0 to machine_recipes.len)
```
Comment (Geevies) [inline code/game/machinery/autolathe.dm]:
```suggestion
		if(index > 0 && index <= machine_recipes.len)
```

File: code/game/machinery/autolathe.dm [hunk 8]
```diff
@@ -184,90 +190,89 @@
...
-		var/datum/autolathe/recipe/making
+		build_item = null
 
 		if(index > 0 && index <= machine_recipes.len)
-			making = machine_recipes[index]
+			build_item = machine_recipes[index]
 
 		//Exploit detection, not sure if necessary after rewrite.
-		if(!making || multiplier < 0 || multiplier > 100)
+		if(!build_item || !(multiplier in 0 to 100))
```
Comment (Geevies) [inline code/game/machinery/autolathe.dm]:
```suggestion
		if(!build_item || multiplier < 0 || multiplier > 100)
```

**Pull request merged**